src/Umbraco.Web/Editors/EntityController.cs

```diff
@@ -235,13 +245,61 @@ private string ParseXPathQuery(string query, int id)
                 },
                 publishedContentExists: i => Umbraco.TypedContent(i) != null);
         }
-        
+
+        #region GetById
+
+        /// <summary>
+        /// Gets an entity by it's id
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="type"></param>
+        /// <returns></returns>
         public EntityBasic GetById(int id, UmbracoEntityTypes type)
         {
             return GetResultForId(id, type);
         }
 
-        public IEnumerable<EntityBasic> GetByIds([FromUri]int[] ids, UmbracoEntityTypes type)
+        /// <summary>
+        /// Gets an entity by it's key
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="type"></param>
+        /// <returns></returns>
+        public EntityBasic GetById(Guid id, UmbracoEntityTypes type)
+        {
+            return GetResultForKey(id, type);
+        }
+
+        /// <summary>
+        /// Gets an entity by it's UDI
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="type"></param>
+        /// <returns></returns>
+        public EntityBasic GetById(Udi id, UmbracoEntityTypes type)
+        {
+            var guidUdi = id as GuidUdi;
+            if (guidUdi != null)
+            {
+                return GetResultForKey(guidUdi.Guid, type);
+            }
+            throw new HttpResponseException(HttpStatusCode.NotFound);
+        } 
+        #endregion
+
+        #region GetByIds
+        /// <summary>
+        /// Get entities by integer ids
+        /// </summary>
+        /// <param name="ids"></param>
+        /// <param name="type"></param>
+        /// <returns></returns>
+        /// <remarks>
+        /// We allow for POST because there could be quite a lot of Ids
+        /// </remarks>
+        [HttpGet]
+        [HttpPost]
+        public IEnumerable<EntityBasic> GetByIds([FromJsonPath]int[] ids, UmbracoEntityTypes type)

```

Comment:
This is the whole point it supports Both as you can see from the attibutes and all of the work i did to make this work for both get/post and all 3 overloads.